ui5/browser/controller/Browser.controller.js

```diff
@@ -283,34 +284,40 @@ sap.ui.define(['sap/ui/core/mvc/Controller',
          return this.websocket.Send("SAVEFILE:" + fullpath + ":" + sText);
       },
 
-      reallyRunMacro: function () {
-         const oEditor = this.getSelectedCodeEditor();
+      reallyRunMacro: function (myThis) {
+         const oEditor = myThis.getSelectedCodeEditor();
          const oModel = oEditor.getModel();
          const fullpath = oModel.getProperty("/fullpath");
          if (fullpath === undefined)
-            return this.onSaveAs();
-         return this.websocket.Send("RUNMACRO:" + fullpath);
+            return myThis.onSaveAs();
+         return myThis.websocket.Send("RUNMACRO:" + fullpath);
       },
 
       /** @brief Handle the "Run" button press event */
       onRunMacro: function () {
+         this.saveCheck(this.reallyRunMacro);
+      },
+
+      saveCheck: function(next) {
```

Comment:
Please use better name for function instead `next`. For instance `PerformSave`.

Reply:
I will find an other name, i called it next because it is the next actions performed by the MessageBox, on some case it run the macro, and other it doesn't to anything
So it is not only for saving, but i will find a better name !

Reply:
Changed it to `functionToRunAfter`